admin/services/headteacher-declaration.service.js

```diff
@@ -18,8 +48,10 @@ headteacherDeclarationService.getEligibilityForSchool = async (dfeNumber) => {
   const pupils = await pupilDataService.sqlFindPupilsWithStatusByDfeNumber(dfeNumber)
   // check the attendance codes for pupils that don't have the completed status
   const ids = pupils.filter(p => p.code !== pupilStatusService.STATUS_CODES.COMPLETED).map(p => p.id)
+  if (ids.length === 0) {
+    return true
```

Comment:
@danwilliams out of curiosity what happens when this returns?

Reply:
@GeorgeChatzigiannis this returns true if all pupils have completed the check (meaning the school is eligible to submit the HDF). If `ids.length > 0` it means some pupils haven’t completed the check and then goes on to check they all have an absent reason.